geomesa-gt/geomesa-gt-spark/src/main/scala/org/locationtech/geomesa/geotools/spark/GeoToolsSpatialRDDProvider.scala

```diff
@@ -0,0 +1,73 @@
+/***********************************************************************
+ * Copyright (c) 2013-2019 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.geotools.spark
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.conf.Configuration
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.geotools.data.{DataStoreFinder, Query, Transaction}
+import org.locationtech.geomesa.spark.{SpatialRDD, SpatialRDDProvider}
+import org.locationtech.geomesa.utils.collection.CloseableIterator
+import org.locationtech.geomesa.utils.geotools.FeatureUtils
+import org.locationtech.geomesa.utils.geotools.converters.FastConverter
+import org.locationtech.geomesa.utils.io.{WithClose, WithStore}
+import org.opengis.feature.simple.SimpleFeature
+
+/**
+  * Spatial RDD provider for arbitrary geotools data stores. If available, prefer a GeoMesa-specific
+  * implementation (e.g. HBaseSpatialRDDProvider), as it will be better optimized.
+  *
+  * To avoid being picked up inadvertently, the data store params must include "geotools" -> "true"
+  */
+class GeoToolsSpatialRDDProvider extends SpatialRDDProvider with LazyLogging {
+
+  import scala.collection.JavaConverters._
+
+  override def canProcess(params: java.util.Map[String, java.io.Serializable]): Boolean = {
+    Option(params.get("geotools")).exists(FastConverter.convert(_, classOf[java.lang.Boolean])) &&
```

Comment:
Is this a change for how one gets an RDD from a GT datastore presently?  (If so, the change might be worth mentioning.)

Reply:
It's not really a change, I think before it just accepted strings which seemed potentially confusing, so I made it also accept Booleans.

Reply:
might want to match with with [JavaGeoMesaSpark](https://github.com/locationtech/geomesa/blob/master/geomesa-spark/geomesa-spark-core/src/main/scala/org/locationtech/geomesa/spark/api/java/JavaGeoMesaSpark.scala#L27), tho, i don't think it will be an issue w/ type erasure

Reply:
hmm, right now SpatialRDDPRovider is `def canProcess(params: java.util.Map[String, java.io.Serializable]): Boolean` and the java API is `def apply(params: java.util.Map[String, _ <: java.io.Serializable])` - probably the java API signature is the one we should go with.

Although I'm not sure where we settled on Serializable, as DataStoreFinder uses `getDataStore(Map params)` (I guess for java 1.4 compat?) and Param uses `lookUp(Map<String, ?> map)`. We should probably make our signatures (java.util.)`Map[String, _]`.